### ISSUE TITLE
Problems Displaying large documents

### ISSUE THREAD

**Issue by GoogleCodeExporter** [fleszar1/flexpaper#80]
```
I am using your viewer on my web site:

http://www.beta.coalstonewcastle.com.au/manual/documents/  

 (click on view option)

What steps will reproduce the problem?
1. Start download and then try and advance page, appears to only go in
twos, ie 1, 3, 5, 7, .....

2. If you advance pages up (before the download is completed and then try
and decrement pages nothing seems to happen).

What is the expected output? What do you see instead?

Not sure whether I am expecting too much as it is a large document. It
almost appears that it is a "buffer" problem, ie too many keystrokes or
mouse clicks are inputted and flash "locks", even though download appears
to still be going.

Is it possible to jump to the last page or middle pages before the document
is downloaded (such as with streaming video)?

The other issue is that it seems to take up a huge amount of memory (almost
1GB according to Win performance monitor). This might be the way flash
operates.

What version of the product are you using? On what operating system?

Flex Paper v 1.3rc1 OS= Win XP SP3

Please provide any additional information below.

I appreciate my comments are a bit vague and rambling. All may be normal,
however I would be interested in your feedback on how you think it copes
with the large file. 

Also is it possible to "jump" to a specific page by inserting a "page
number" in the calling "url"? Similar to pdf documents that can be opened
at a specific page.

I also want to thank you for your work, I am looking forward to using it
further in my web site.
```

Original issue reported on code.google.com by `coals1...@gmail.com` on 25 May 2010 at 12:10

**Comment by GoogleCodeExporter** [comment]
```
Hi,
This issue is twofold;
The first comes from the web server not returning the content-length on the SWF 
file 
which causes FlexPaper to go to +Infinity when its trying to calculate the 
remaining 
bytes. I will create a solution for this and package that into the final 
release of 1.3

The second problem comes from swf file itself (the converted document) where it 
seems to 
contain quite a few images causing a memory leakage inside the flash player. It 
would be 
interesting to see if this problem persists in the latest version of PDF2SWF 
because I 
know other people have had problems with large documents containing lots of 
images in 
previous versions of PDF2SWF. Could you try converting it with the latest 
release of 
PDF2SWF and upload the new file onto here?
```

Original comment by `erik.eng...@gmail.com` on 27 May 2010 at 4:43

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the feedback.

Just to clarify item 2 - I am using pdf2swf v0.9, do you want me to try 
converting it
using a beta version?
```

Original comment by `coals1...@gmail.com` on 28 May 2010 at 11:00

**Comment by GoogleCodeExporter** [comment]
```
Or if you could post the PDF file which you converted and I will do a few tests 
with it
- thanks

Erik
```

Original comment by `erik.eng...@gmail.com` on 30 May 2010 at 8:23

**Comment by GoogleCodeExporter** [comment]
```
Raw PDF file attached. - Can you please download and remove from public 
display. Thanks
```

Original comment by `coals1...@gmail.com` on 30 May 2010 at 9:14

**Comment by GoogleCodeExporter** [comment]
```
Similar problem.
Large file - 39.7 megs
not displaying
after frame appears it stops loading/downloading
Also I am posting behind an https connection, which might have an impact 
(negative,
that is)
Maybe I have a file naming problem? Making mistakes is possible. I'm a newbie 
at this.
Is it really necessary to have everything called FlexPaperViewer or "paper" - 
as in
the test document?
Manny thanks
```

Original comment by `cheapglobalmobile@gmail.com` on 30 May 2010 at 9:20

**Comment by GoogleCodeExporter** [comment]
```
Yes, I also have problems with big pdf files:
http://www.2shared.com/file/UiJ97_X-/31_online.html - original pdf (~30mb)
http://www.2shared.com/video/9Wehzuug/31_online.html - converted swf

Pictures from the center of the book disappear when zooming in :(


```

Original comment by `igba...@gmail.com` on 11 Jun 2010 at 9:13

**Comment by GoogleCodeExporter** [comment]
```
I had another go at the example with the country time table in the latest 
version of flash and it actually seems much better managed now memory wize. It 
still allocates quite a lot of memory but seems to release memory in a much 
better fashion. 

Please try again with latest version of Flash and latest version of FlexPaper
```

Original comment by `erik.eng...@gmail.com` on 26 Jun 2010 at 9:43

**Comment by GoogleCodeExporter** [comment]
```
We have a solution coming up for very large documents. Details about this will 
be released in the next few weeks. 
```

Original comment by `erik.eng...@devaldi.com` on 16 Oct 2010 at 11:57

**Comment by GoogleCodeExporter** [comment]
```
Has this solution been released ?
```

Original comment by `joseph.m...@gmail.com` on 6 Nov 2010 at 1:45

**Comment by GoogleCodeExporter** [comment]
```
We are still working on the split-up page solution. It is still very much a 
priority to get released but we are quite keen on getting the quality to a 
level we are comfortable with. 

In the meantime, I would recommend anyone who is trying to convert large 
documents to add the following flag to their conversion: "-s subpixels=1.5"

An example of a complete set of recommended parameters using PDF2SWF would be:
C:\SWFTools\pdf2swf.exe Paper.pdf -o Paper.swf -f -T 9 -t -G -s 
storeallcharacters -s subpixels=1.5
```

Original comment by `erik.eng...@devaldi.com` on 11 Nov 2010 at 7:12

**Comment by GoogleCodeExporter** [comment]
```
And by large I mean image heavy documents. Normal documents, even with many 
pages does not need this flag.
```

Original comment by `erik.eng...@devaldi.com` on 11 Nov 2010 at 7:16

**Comment by GoogleCodeExporter** [comment]
```
Hello Erik - is this new release out with the fixes for the large documents and 
images. Thanks. Anand
```

Original comment by `anandamr...@gmail.com` on 14 Jan 2011 at 10:48

**Comment by GoogleCodeExporter** [comment]
```
Hello Erik, I have get the issue when I set the ProgressiveLoading true.
```

Original comment by `huxia...@gmail.com` on 31 Mar 2011 at 7:20

**Comment by GoogleCodeExporter** [comment]
```
Hi,
Solution for large documents is coming up. Most of the source code has been 
committed and the last bits and pieces are being reviewed in the next few days.
```

Original comment by `erik.eng...@devaldi.com` on 12 May 2011 at 6:08
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Fixed by the ability to load split-up documents. Implemented in 1.4.5 which is 
now available for download 
```

Original comment by `erik.eng...@devaldi.com` on 23 May 2011 at 8:27
- Changed state: **Fixed**
